test/timetable.test.js

```diff
@@ -0,0 +1,197 @@
+    },
+  ]
+
+  const result = tt.calculateHourLabels(timeline)
+  const rlen = result.length
+  const resultSelection = [result[0], result[1], result[rlen - 2], result[rlen - 1]]
+
+  t.deepEqual(resultSelection, expected, 'calculate hour labels for timeline 7:30 - 21:15')
+
+  t.end()
+})
+
+test('mapPropertiesToClass()', t => {
+  const properties = {
+    isA: true,
+    isB: false,
+    isProperty: true,
+    isLoremIpsum: false,
+    isDolorSitAmet: true,
+  }
+
+  const expected = 'class class--a class--property class--dolorsitamet '
+
+  t.equal(tt.mapPropertiesToClass(properties, 'class'), expected, 'returns properties transformed to classname')
+  t.end()
+})
+
+test('groupEventsByDays()', t => {
+  const events = [
+    {startsAt: '2015-12-03 12:00:00'},
+    {startsAt: '2015-12-03 14:00:00'},
+    {startsAt: '2015-12-04 12:00:00'},
+    {startsAt: '2015-12-05 12:00:00'},
+  ]
+
+  const expected = {
+    3: [
+      {startsAt: '2015-12-03 12:00:00'},
+      {startsAt: '2015-12-03 14:00:00'},
+    ],
+    4: [
+      {startsAt: '2015-12-04 12:00:00'},
+    ],
+    5: [
+      {startsAt: '2015-12-05 12:00:00'},
+    ],
+  }
+
+  t.deepEqual(tt.groupEventsByDays(events), expected, 'groups events by weekdaynum')
+
+  t.end()
+})
+
+test('calculateOverlap()', t => {
+  const expected = [
+    {startsAt: '2015-12-03 12:00:00', endsAt: '2015-12-03 13:00:00', _firstOverlapping: false, _overlaps: 0},
+    {startsAt: '2015-12-03 13:00:00', endsAt: '2015-12-03 14:00:00', _firstOverlapping: false, _overlaps: 0},
+    {startsAt: '2015-12-03 15:00:00', endsAt: '2015-12-03 17:00:00', _firstOverlapping: false, _overlaps: 0},
+    {startsAt: '2015-12-03 20:00:00', endsAt: '2015-12-03 20:30:00', _firstOverlapping: false, _overlaps: 0},
+    {startsAt: '2015-12-04 12:00:00', endsAt: '2015-12-04 13:00:00', _firstOverlapping: true, _overlaps: 1},
+    {startsAt: '2015-12-04 12:30:00', endsAt: '2015-12-04 14:00:00', _firstOverlapping: false, _overlaps: 1},
+    {startsAt: '2015-12-04 15:00:00', endsAt: '2015-12-04 17:00:00', _firstOverlapping: true, _overlaps: 1},
+    {startsAt: '2015-12-04 16:00:00', endsAt: '2015-12-04 17:00:00', _firstOverlapping: false, _overlaps: 1},
+    {startsAt: '2015-12-05 12:00:00', endsAt: '2015-12-05 13:00:00', _firstOverlapping: true, _overlaps: 2},
+    {startsAt: '2015-12-05 12:30:00', endsAt: '2015-12-05 13:00:00', _firstOverlapping: false, _overlaps: 2},
+    {startsAt: '2015-12-05 12:45:00', endsAt: '2015-12-05 15:00:00', _firstOverlapping: false, _overlaps: 2},
+    {startsAt: '2015-12-05 15:00:00', endsAt: '2015-12-05 20:30:00', _firstOverlapping: false, _overlaps: 0},
+  ]
+
+  const events = R.map(({startsAt, endsAt}) => {
+    return {
+      startsAt,
+      endsAt,
+    }
+  }, expected)
+
+  const overlaps = tt.calculateOverlap(events)
+
+  if (events.length !== overlaps.length) {
+    t.fail('length of events and overlapped events have to be equal')
+  }
+
+  let eventid
+  for ( eventid in events ) {
```

Comment:
Actually, I think you could just use `deepEqual` on both arrays instead of iterating them. Don't you think?